smartsim/_core/mli/infrastructure/storage/featurestore.py

```diff
@@ -24,15 +24,39 @@
 # OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
 # OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
+import enum
 import typing as t
 from abc import ABC, abstractmethod
 from dataclasses import dataclass
 
+from smartsim.error.errors import SmartSimError
 from smartsim.log import get_logger
 
 logger = get_logger(__name__)
 
 
+class ReservedKeys(str, enum.Enum):
+    """Contains constants used to identify all featurestore keys that
+    may not be to used by users. Avoids overwriting system data"""
+
+    MLI_NOTIFY_CONSUMERS = "_SMARTSIM_MLI_NOTIFY_CONSUMERS"
+    """Storage location for the list of registered consumers that will receive
+    events from an EventBroadcaster"""
+
+    @classmethod
+    def from_string(cls, value: str) -> t.Optional["ReservedKeys"]:
+        """Convert a string representation into an enumeration member
+
+        :param value: the string to convert
+        :returns: the enumeration member if the conversion succeeded, otherwise None"""
```

Comment:
When would this fail?

Reply:
any string not in the enumeration. see `test_mli_reserved_keys_conversion`